### PR TITLE
Enable Papertrail logging using remote_syslog and supervisord

### PR DESCRIPTION
**Pull request by dgl-ableton**

### CODE REVIEW

**Comment by nre-ableton** [comment]
@ablbot autosquash

**Comment by dgl-ableton** [comment]
@ablbot autosquash

**Comment by dgl-ableton** [comment]
@ablbot autosquash

**Comment by ghost** [comment]
Rebase job `2111e951-0506-444a-9b43-e4acad238071` has been created. There are currently 0 job(s) in the rebaser queue.

**Comment by dgl-ableton** [comment]
@ablbot autosquash

**Comment by dgl-ableton** [comment]
Thx @nre-ableton for the help! (Sorry your name got deleted 😞 )

**Comment by dgl-ableton** [comment]
👋 @AbletonAppDev/gotham-city

File: Dockerfile [hunk 1]
```diff
@@ -6,13 +6,25 @@ VOLUME /output
 RUN mkdir -p /jenkins_node_scanner
 WORKDIR /jenkins_node_scanner
 
+RUN apt-get update \
+    && apt-get install -y supervisor=3.3.5-1 wget=1.20.1-1.1 --no-install-recommends \
+    && wget -q -O - https://github.com/papertrail/remote_syslog2/releases/download/v0.17/remote_syslog_linux_amd64.tar.gz \
```
Comment (nre-ableton) [inline Dockerfile]: I don't think that the `wget` and the `apt-get` stuff should be on the same stripe. Also, it makes the diff look weird a few commits later on. Please move this to a separate `RUN` command.

File: Dockerfile [hunk 2]
```diff
@@ -6,13 +6,25 @@ VOLUME /output
...
+
+
 COPY Pipfile /jenkins_node_scanner
 COPY Pipfile.lock /jenkins_node_scanner
 RUN pip install --no-cache-dir pipenv==2018.11.26
 RUN pipenv install --system --ignore-pipfile
 
+RUN mkdir -p /logs/server
+
+# Copy the supervisor config file
```
Comment (nre-ableton) [inline Dockerfile]: This comment doesn't add much info, please remove.

**Review by nre-ableton — CHANGES_REQUESTED** [review]
Commit ccf985ed7d1cfadad8f551b5505039a2158ab317 is weird, and it won't work in isolation. Please squash it to d89c5dba6f46d13f4cc48c04850a1c9abd08af28. Also there's a typo in the PR title -- it should be "Enable _papertrail_ logging..."

**Comment by dgl-ableton** [comment]
@ablbot autosquash

File: Dockerfile [hunk 3]
```diff
@@ -6,13 +6,25 @@ VOLUME /output
...
+    && rm -rf /var/lib/apt/lists/*
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN wget -q -O - https://github.com/papertrail/remote_syslog2/releases/download/v0.17/remote_syslog_linux_amd64.tar.gz \
+    | tar -zxf -
+
+
 COPY Pipfile /jenkins_node_scanner
 COPY Pipfile.lock /jenkins_node_scanner
+COPY jenkins_node_scanner.py /jenkins_node_scanner
```
Comment (nre-ableton) [inline Dockerfile]: Why did this line move up here? Let's move it back to be next to the other `COPY` statements.

File: Jenkinsfile [hunk 4]
```diff
@@ -8,6 +8,14 @@ devToolsProject.run(
     sh 'pipenv sync --dev'
   },
   build: { data ->
+    Map creds = encryptedFile.readJson(
+      path: 'credentials.json.enc',
+      credentialsId: 'jenkins-node-scanner-password',
+    )
+    String jinjaCommand = "jinja2 -D host=${creds['host']} -D port=${creds['port']}"
+    sh "pipenv run ${jinjaCommand} -o papertrail_config.yml papertrail_config.yml.j2"
```
Comment (nre-ableton) [inline Jenkinsfile]: For future reference, here's a nifty Groovy trick you could use in the future:

```groovy
['papertrail_config.yml', 'supervisord.conf'].each { i ->
  sh "pipenv run ${jinjaCommand} -o ${i} ${i}.j2"
}
```

That's probably overkill when we only have 2 files here, but if we had a bunch more then you'd definitely want to use an `each` loop to save copy/pasting. Anyways, you can take or leave that suggestion as far as this commit goes.

**Review by nre-ableton — CHANGES_REQUESTED** [review]
First of all, the commit history is looking _much_ better. Thanks for clearing that up. There's a few things that I missed from the first round, though:

- `supervisord.conf` and `papertrail_config.yml` should **not** be added in a08b360e16cace9c0bde997db63d5bce1eaa9952. The point of that commit is to generate them from Jinja2 templates. Please remove both of these files from git altogether.
- Please 2 spaces when indenting YAML files (in this case, only `papertrail_config.yml.j2`

File: Dockerfile (hunk 3 above)
Comment (dgl-ableton) [reply Dockerfile]: Ah my bad! I thought it would make more sense for that line to go there since I didn't think it was Papertrail specific. Thanks!

**Comment by dgl-ableton** [comment]
@ablbot autosquash

**Review by nre-ableton — CHANGES_REQUESTED** [review]
`supervisord.conf` is still present

**Comment by dgl-ableton** [comment]
@ablbot autosquash

**Comment by nre-ableton** [comment]
@dgl-ableton The indentation changes to the YAML file got lost during rebasing, I pushed a fixup just FYI. Gonna rebase now.

**Comment by nre-ableton** [comment]
@ablbot autosquash

**Review by nre-ableton — APPROVED** [review]
:shipit:

**Pull request merged**